qiskit/backends/ibmq/ibmqjob.py

```diff
@@ -356,16 +434,17 @@ def _reorder_bits(result):
     """temporary fix for ibmq backends.
     for every ran circuit, get reordering information from qobj
     and apply reordering on result"""
-    for idx, circ in enumerate(result._qobj['circuits']):
-
+    for circuit_result in result._result['result']:
+        if 'metadata' in circuit_result:
+            circ = circuit_result['metadata'].get('compiled_circuit')
+        else:
+            raise QISKitError('result object missing metadata for reordering bits')
```

Comment:
This get raised when calling `job.result()` for all jobs that were run prior to the new `metadata` field, which seems quite recent. Should it at least return something, even though it could potentially miss some reorderings? Otherwise this PR is not immediately usable if one goes searching for past jobs.

Reply:
Ok, we can return and log warning.